SelfAndPeerEvaluations/utils.py

```diff
@@ -0,0 +1,439 @@
+            )
+            self.scores[qualitative_subject + "Self"].append(self_score)
+            self.scores[qualitative_subject + "Average"].append(self_score)
+            self.scores[qualitative_subject + "Diff"].append(0)
+        row.append(0)  # average qualitative difference
+        self.scores["Qualitative Difference"].append(0)
+        self_score = json_file[self.id]["Project Contribution"][0]
+        row.append(
+            f"Self: {self_score}\nPartner: N/A\nAverage: {self_score}\nDiff (self - partner): 0"
+        )
+        self.scores["Project Contribution" + "Self"].append(self_score)
+        self.scores["Project Contribution" + "Average"].append(self_score)
+        self.scores["Project Contribution" + "Diff"].append(0)
+
+    def __give_quiz_scores(self, json_file: dict, row: list) -> None:
+        qualitative_differences = []
+        for qualitative_subject in json_file["info"]["qualitative_subjects"]:
+            self_score = json_file[self.id][qualitative_subject][0]
+            partner_score = json_file[self.id][qualitative_subject][1]
+            average_score = round(
+                statistics.fmean(json_file[self.id][qualitative_subject]), 2
+            )
+            qualitative_difference = self_score - partner_score
+            qualitative_differences.append(qualitative_difference)
+            row.append(
+                f"Self: {self_score}\nPartner: {partner_score}\nAverage: {average_score}\nDiff (self - partner): {qualitative_difference}"
+            )
+            self.scores[qualitative_subject + "Self"].append(self_score)
+            self.scores[qualitative_subject + "Partner"].append(partner_score)
+            self.scores[qualitative_subject + "Average"].append(average_score)
+            self.scores[qualitative_subject + "Diff"].append(qualitative_difference)
+        row.append(
+            round(statistics.fmean(qualitative_differences), 2)
+        )  # average qualitative difference
+        self.scores["Qualitative Difference"].append(
+            round(statistics.fmean(qualitative_differences), 2)
+        )
+        self_score = json_file[self.id]["Project Contribution"][0]
+        partner_score = json_file[self.id]["Project Contribution"][1]
+        average_score = round(
+            statistics.fmean(json_file[self.id]["Project Contribution"]),
+            2,
+        )
+        contrib_diff = self_score - partner_score
+        row.append(
+            f"Self: {self_score}\nPartner: {partner_score}\nAverage: {average_score}\nDiff (self - partner): {contrib_diff}"
+        )
+        self.scores["Project Contribution" + "Self"].append(self_score)
+        self.scores["Project Contribution" + "Partner"].append(partner_score)
+        self.scores["Project Contribution" + "Average"].append(average_score)
+        self.scores["Project Contribution" + "Diff"].append(contrib_diff)
+
+    def __read_files(self, files: list) -> None:
```

Comment:
Improve type hint for files.
```suggestion
    def __read_files(self, files: list[str]) -> None:
```